netbox/netbox/views/generic/bulk_views.py

```diff
@@ -321,13 +321,47 @@ def clean(self):
 
         return ImportForm(*args, **kwargs)
 
-    def _create_objects(self, form, request):
-        new_objs = []
+    def _get_records(self, form, request):
         if request.FILES:
             headers, records = form.cleaned_data['csv_file']
         else:
             headers, records = form.cleaned_data['csv']
 
+        return headers, records
+
+    def _update_objects(self, form, request, headers, records):
+        from utilities.forms import CSVModelChoiceField
+        new_objs = []
+
+        for row, data in enumerate(records, start=1):
+            try:
+                obj = self.queryset.model.objects.get(pk=data["id"])
```

Comment:
We should extract the IDs list and retrieve all of the instances in a single query. We can create a dictionary mapping each ID to its instance and quickly check for any missing keys.